### ISSUE TITLE
multi-layer feature aggregation strategy and long-term tracking

### ISSUE THREAD

**Issue by zzzmm1** [MCG-NJU/MixFormer#25]
Thanks for sharing the excellent work!
I have two small questions.
First, as mentioned in your paper,  the multi-layer feature aggregation strategy is commonly used in other trackers (e.g., SiamRPN++, STARK). The one in SiamRPN++ is understandable, but the one in STARK is confusing. STARK seems to only use the last stride=16 features for prediction.  I would like to know what is the main difference between MixFormer and STARK in this regard?
Second, have you tested the MixFormer on the VOT long-term dataset? STARK performs well on long-term tracking, and it feels like MixFormer could work better.

**Comment by yutaocui** [comment]
Thanks for your questions. For the first one, I have misunderstood the code of STARK. The following tag is set to 'True' only in mask prediction head: 
https://github.com/researchmm/Stark/blob/4071d9b940fdb853921604cad48128415aa8665a/lib/models/stark/backbone.py#L138
In fact, it also doesn't use multi-layer feature aggregation strategy. We will correct the paper in the following version.
By the way, we have tested the trick in MixFormer and obtained a higher AUC on LaSOT/Tracking.
For the second one, we haven't evaluated MixFormer on VOT-LT and you can have a try.

**Comment by zzzmm1** [comment]
Thanks for your answer.
If I understand correctly, when using the multi-layer feature aggregation strategy, it is necessary to perform prediction and weighted fusion on feature maps of different sizes and needs to adjust the overall network structure and retrain, instead of modifying the testing process directly on the pretrain model.
Will this cause a significant drop in tracking speed?
For long-term tracking, I have tested MixFormer-22k on the VOTLT dataset, and it seems that there is no obvious advantage compared to STARK-ST. Do you have any suggestions?
Thanks a lot!

**Comment by yutaocui** [comment]
Yes, it's necessary to train a new model when using the multi-layer feature aggregation strategy. We only concatenate feature maps of different layers and reduce the channel with a linear function, so that it has little influence on tracking speed.
For VOT-LT, maybe you can try to adjust the hyper-parameters of `UPDATE_INTERVAL` and `ONLINE_SIZES`.

**Comment by yutaocui** [comment]
> Thanks for your answer.
> If I understand correctly, when using the multi-layer feature aggregation strategy, it is necessary to perform prediction and weighted fusion on feature maps of different sizes and needs to adjust the overall network structure and retrain, instead of modifying the testing process directly on the pretrain model.
> Will this cause a significant drop in tracking speed?
> For long-term tracking, I have tested MixFormer-22k on the VOTLT dataset, and it seems that there is no obvious advantage compared to STARK-ST. Do you have any suggestions?
> Thanks a lot!

I think you should first use the hyper-parameter of mixformer-lasot since it's also a long-term dataset.

**Comment by zzzmm1** [comment]
Thanks for your suggestion!
I will try to adjust the hyper-parameters.

**Comment by zzzmm1** [comment]
Hi, I haved tested the pretrain MixFormer-22k on LaSOT and got the following results:
lasot                | AUC        | OP50       | OP75       | Precision    | Norm Precision    |
MixFormerOnline      | 68.67      | 80.45      | 67.71      | 74.58        | 78.69             |
It is slightly different from the downloaded raw results.  Are there any details I overlooked?
Is it correct that the search_area_scale in test_mixformer.sh is 4.55? Because it is 4.5 in other datasets and 5.0 in STARK.
My device is 2080Ti, and the environment is Torch 1.7.0+cu101 and torchvision 0.8.1+cu101.
